src/app/routes/enrolment/view-requests/view-requests.component.ts

```diff
@@ -167,13 +169,26 @@ export class ViewRequestsComponent implements OnInit {
     if (this.claim.token) {
       const decoded = await this.decode(this.claim.token);
       if (decoded.claimData) {
-        this.fields = decoded.claimData?.fields
-          ? decoded.claimData?.fields
-          : [];
+        this.requestorFields = this.getRequestorFields(decoded.claimData);
       }
     }
   }
 
+  private getRequestorFields(data: {
+    requestorFields?: KeyValue<string, string | number>[];
```

Comment:
Maybe we can extract this this, ex `type FieldsKeyValue = KeyValue<string, string | number>`
```suggestion
    requestorFields?: FieldsKeyValue[];
```
I saw that we using it in couple of places

Reply:
Good idea

Reply:
Do you want to change it in this PR or later?

Reply:
is iam-client-lib adjusted with those changes? If yes, then i will do it in this PR